src/qtils/macro.hpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#define IDENTITY(a) a
+
+#define CONCAT__(a, b) a##b
+#define CONCAT_(a, b) CONCAT__(a, b)
+#define CONCAT(a, b) CONCAT_(IDENTITY(a), IDENTITY(b))
+
+#define QTILS_UNIQUE_NAME(prefix) CONCAT(prefix, __COUNTER__)
```

Comment:
Duplicates `OUTCOME_UNIQUE` from <qtils/outcome.hpp>

May move macros to separate files ("unique name" and "unexpected" are not related)

```suggestion
#define _QTILS_UNIQUE_NAME_2(x, y) x##y
#define _QTILS_UNIQUE_NAME(prefix, counter) _QTILS_UNIQUE_NAME_2(prefix, counter)
#define QTILS_UNIQUE_NAME_PREFIX(prefix) _QTILS_UNIQUE_NAME(prefix, __COUNTER__)
#define QTILS_UNIQUE_NAME QTILS_UNIQUE_NAME_PREFIX(qtils_unique_name_)
```

Reply:
Ah, I thought I saw it somewhere else.